### ISSUE TITLE
💅 `noRestrictedImports` does not flag imports that don't match exactly

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#6233]
### Environment information

```bash
CLI:
  Version:                      2.0.0-beta.6
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v18.20.3
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.9

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            false
  Linter enabled:               true
  Assist enabled:               false
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    style/noRestrictedImports

Workspace:
  Open Documents:               0
```

### Rule name

style/noRestrictedImports

### Playground link

https://github.com/dyc3/biome-repro-1749131484111

### Expected result

noRestrictedImports should support our glob syntax for matching import names.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Oh, duh 🤦 I've looked at that PR so many times...
